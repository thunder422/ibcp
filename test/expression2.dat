###################################################
#  Expression Test #2: Parenthetical Expressions  #
###################################################

(A + B) + C
(A * B) + C
(A + B) * C
A + (B + C)
A + (B + C) + D
A + (B + C) * D
(A + (B + C)) + D
(A + (B + C)) * D
A + (B * C)
A + (B * C) + D
A + (B * C) * D
A + (B * C) ^ D
A + ((B * C) + D)
A + ((B * C) * D)
A + ((B * C) ^ D)
A * (B + C)
A * (B + C) + D
A * (B + C) * D
A + (B) + C
A + ((B)) + C
A + B + ((C))
(A + B + C)
((A + B)) + C
A + ((B * C))
A + B * C ^ D
((A + B) * C) ^ D
-A ^ 2 + (-B) ^ 3
(A + B) * (C + D)
(A + B) + (C + D)
A + B + (C + D)
(A * B) + (C * D)
(A + (B * (C ^ D))) + E
(A ^ (B * (C + D))) + E
# test missing close parentheses
(A + B
# test missing open parentheses
A + B)
# more unary operator tests
-NOT A%
NOT -A%
-A + B
-(A + B)
-A ^ B
# more parentheses tests
A + B + (((C)))
(((A + B))) + C
((((A + B)))) + C
