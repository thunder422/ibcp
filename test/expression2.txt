
Input: (A + B) + C
Output: A B +')' C + 

Input: (A * B) + C
Output: A B *')' C + 

Input: (A + B) * C
Output: A B + C * 

Input: A + (B + C)
Output: A B C + + 

Input: A + (B + C) + D
Output: A B C + + D + 

Input: A + (B + C) * D
Output: A B C + D * + 

Input: (A + (B + C)) + D
Output: A B C + +')' D + 

Input: (A + (B + C)) * D
Output: A B C + + D * 

Input: A + (B * C)
Output: A B C *')' + 

Input: A + (B * C) + D
Output: A B C *')' + D + 

Input: A + (B * C) * D
Output: A B C *')' D * + 

Input: A + (B * C) ^ D
Output: A B C * D ^ + 

Input: A + ((B * C) + D)
Output: A B C *')' D + + 

Input: A + ((B * C) * D)
Output: A B C *')' D *')' + 

Input: A + ((B * C) ^ D)
Output: A B C * D ^')' + 

Input: A * (B + C)
Output: A B C + * 

Input: A * (B + C) + D
Output: A B C + * D + 

Input: A * (B + C) * D
Output: A B C + * D * 

Input: A + (B) + C
Output: A B')' + C + 

Input: A + ((B)) + C
Output: A B')' ) + C + 

Input: A + B + ((C))
Output: A B + C')' ) + 

Input: (A + B + C)
Output: A B + C +')' 

Input: ((A + B)) + C
Output: A B +')' ) C + 

Input: A + ((B * C))
Output: A B C *')' ) + 

Input: A + B * C ^ D
Output: A B C D ^ * + 

Input: ((A + B) * C) ^ D
Output: A B + C * D ^ 

Input: -A ^ 2 + (-B) ^ 3
Output: A 2 ^* Neg B Neg 3 ^* + 

Input: (A + B) * (C + D)
Output: A B + C D + * 

Input: (A + B) + (C + D)
Output: A B +')' C D + + 

Input: A + B + (C + D)
Output: A B + C D + + 

Input: (A * B) + (C * D)
Output: A B *')' C D *')' + 

Input: (A + (B * (C ^ D))) + E
Output: A B C D ^')' *')' +')' E + 

Input: (A ^ (B * (C + D))) + E
Output: A B C D + * ^')' E + 

Input: (A + B
             ^-- expected operator or closing parentheses


Input: A + B)
            ^-- expected operator or end-of-expression


