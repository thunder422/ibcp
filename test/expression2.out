
Input: (A + B) + C
Output: TEST:A B +')' C + 

Input: (A * B) + C
Output: TEST:A B *')' C + 

Input: (A + B) * C
Output: TEST:A B + C * 

Input: A + (B + C)
Output: TEST:A B C + + 

Input: A + (B + C) + D
Output: TEST:A B C + + D + 

Input: A + (B + C) * D
Output: TEST:A B C + D * + 

Input: (A + (B + C)) + D
Output: TEST:A B C + +')' D + 

Input: (A + (B + C)) * D
Output: TEST:A B C + + D * 

Input: A + (B * C)
Output: TEST:A B C *')' + 

Input: A + (B * C) + D
Output: TEST:A B C *')' + D + 

Input: A + (B * C) * D
Output: TEST:A B C *')' D * + 

Input: A + (B * C) ^ D
Output: TEST:A B C * D ^ + 

Input: A + ((B * C) + D)
Output: TEST:A B C *')' D + + 

Input: A + ((B * C) * D)
Output: TEST:A B C *')' D *')' + 

Input: A + ((B * C) ^ D)
Output: TEST:A B C * D ^')' + 

Input: A * (B + C)
Output: TEST:A B C + * 

Input: A * (B + C) + D
Output: TEST:A B C + * D + 

Input: A * (B + C) * D
Output: TEST:A B C + * D * 

Input: A + (B) + C
Output: TEST:A B')' + C + 

Input: A + ((B)) + C
Output: TEST:A B')' ) + C + 

Input: A + B + ((C))
Output: TEST:A B + C')' ) + 

Input: (A + B + C)
Output: TEST:A B + C +')' 

Input: ((A + B)) + C
Output: TEST:A B +')' ) C + 

Input: A + ((B * C))
Output: TEST:A B C *')' ) + 

Input: A + B * C ^ D
Output: TEST:A B C D ^ * + 

Input: ((A + B) * C) ^ D
Output: TEST:A B + C * D ^ 

Input: -A ^ 2 + (-B) ^ 3
Output: TEST:A 2% ^* Neg B Neg 3% ^* + 

Input: (A + B) * (C + D)
Output: TEST:A B + C D + * 

Input: (A + B) + (C + D)
Output: TEST:A B +')' C D + + 

Input: A + B + (C + D)
Output: TEST:A B + C D + + 

Input: (A * B) + (C * D)
Output: TEST:A B *')' C D *')' + 

Input: (A + (B * (C ^ D))) + E
Output: TEST:A B C D ^')' *')' +')' E + 

Input: (A ^ (B * (C + D))) + E
Output: TEST:A B C D + * ^')' E + 

Input: (A + B
             ^-- expected operator or closing parentheses

Input: A + B)
            ^-- expected operator or end-of-statement

